### ISSUE TITLE
ngrx V6: upgrade to rxjs 6 and ts2.7

### ISSUE THREAD

**Issue by sandangel** [ngrx/platform#926]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[x] Feature request
[ ] Documentation issue or request
</code></pre>

## What is the current behavior?

Could not compile app with ngrx, Angular v6.0.0-rc.0, rxjs and rxjs-compat v6.0.0-beta.1

```bash
ERROR in ../node_modules/@ngrx/effects/@ngrx/effects.es5.js
Module not found: Error: Can't resolve 'rxjs/operator/exhaustMap' in ...
```

## Expected behavior:

I hope I could upgrade my app to Angular v6 by the time it releases. But I'm using a lot of ngrx in my code and it seems not compatible with Angular v6 and ts 2.7 yet.

## Minimal reproduction of the problem with instructions:

Create an new cli project and run `ng update --next`
then run `ng build`

## Version of affected browser(s),operating system(s), npm, node and ngrx:
ngrx v5.2.0, node 8.10.0, yarn 1.5.1
## Other information:
Please tell me if I could help. I'm willing to do a PR.

**Comment by MikeRyanDev** [comment]
@brandonroberts and I plan on taking a stab at this on March 30th. If someone else wants to try they are welcome.

We'll accept PRs before then in two steps:
1. Update to TypeScript 2.7. Don't worry about trying to get the Bazel build to pass.
2. Update to RxJS 6.0. This will be a heavy refactor as you'll want to change to pipeable operators in all of the packages.

**Comment by sandangel** [comment]
@MikeRyanDev Thanks, I will give it a try. Do you want to turn on strict mode property initialize in ts2.7?

**Comment by brandonroberts** [comment]
Fixed via d1286d221ec958a08b306ae85bf9d68ea267a40d
